hpack/hpack.py

```diff
@@ -231,19 +234,20 @@ def encode(self, headers, huffman=True):
         if isinstance(headers, dict):
             headers = headers.items()
 
-        # Next, walk across the headers and turn them all into bytestrings.
-        headers = [(_to_bytes(n), _to_bytes(v)) for n, v in headers]
-
         # Before we begin, if the header table size has been changed we need
         # to signal that appropriately.
         if self._table_size_changed:
             header_block.append(self._encode_table_size_change())
             self._table_size_changed = False
 
-        # We can now encode each header in the block.
-        header_block.extend(
-            (self.add(header, huffman) for header in headers)
-        )
+        # Add each header to the header block
+        for h in headers:
+            to_add = (_to_bytes(h[0]),_to_bytes(h[1]))
+            if(len(h) == 2):
+                header_block.extend(self.add(to_add,False,huffman))
+            else:
+                header_block.extend(self.add(to_add,h[2],huffman))
```

Comment:
Rather than pass a mystery-meat parameter to `add`, let's assign `h[2]` to a temporary variable:

```python
never_index = h[2] if len(h) > 2 else False
```